hicommonwealth/commonwealth
# Added recovery migration script to fix view counts

## Link to Issue
Closes: #5591

## Description of Changes
- Adds a new column new_view_counts which will track the regular view_counts
- Adds a script which will migrate view counts from a 2nd recovery db containing the old view counts

## Test Plan
High level:
- Set up recovery DB with old db dump
- Run migration script on main DB.
- Run script in 1 off dyno

Low level steps:
1. Start up a docker container with an old dump (older than root-id-refactor) on a different port with:
```docker run --name your_postgres_container -e POSTGRES_USER=commonwealth -e POSTGRES_DB=commonwealth -e POSTGRES_PASSWORD=edgeware -p 5433:5432 -d postgres```
2. update sequelize.json to point to port 5433 (recovery db)
3. Set up this db. Run db-all on this DB, as well as all migration scripts on this up to the root-id-refactor. I like to do this by adding `fail` to the root-id-refactor up statement. Then when running db-all, it fails at this script and stops.
4. Add to environment variables: `RECOVERY_DATABASE_URI=postgres://commonwealth:edgeware@localhost:5433/commonwealth`
5. update sequelize.json to point to port 5432 (main db)
4. On main db, run migration script.
5. Run one off dyno with ts-node recoverViewCounts.ts
6. Check to make sure view counts are correct

## Deployment Plan
1. Merge
2. Add RECOVERY_DB_URI env to production (db already set up, just need to add the env var)
3. heroku run bash -a [heroku app here]
4. cd packages/commonwealth
5. yarn run-view-counts
---- 
After running this deployment plan, we have a followup PR that will remove this extra field:
https://github.com/hicommonwealth/commonwealth/pull/6132
After merging in 6132, the view counts will be fixed, and the site will be updated. 5911 is just to prepare the data ready in a safe and reversible way.
